src/base/bittorrent/session.cpp

```diff
@@ -3984,18 +3982,34 @@ void Session::handleTorrentRemovedAlert(libt::torrent_removed_alert *p)
 {
     if (m_loadedMetadata.contains(p->info_hash))
         emit metadataLoaded(m_loadedMetadata.take(p->info_hash));
+
+    if (m_removedTorrents.contains(p->info_hash)) {
+        const RemovedTorrentData tmpRemovedTorrentData = m_removedTorrents[p->info_hash];
+        if (!tmpRemovedTorrentData.deletesFiles) {
+            LogMsg(tr("'%1' was removed from the transfer list.", "'xxx.avi' was removed...").arg(tmpRemovedTorrentData.name));
+            m_removedTorrents.remove(p->info_hash);
+        }
+    }
 }
 
 void Session::handleTorrentDeletedAlert(libt::torrent_deleted_alert *p)
 {
-    Utils::Fs::smartRemoveEmptyFolderTree(m_savePathsToRemove.take(p->info_hash));
+    const RemovedTorrentData tmpRemovedTorrentData = m_removedTorrents.take(p->info_hash);
+    Utils::Fs::smartRemoveEmptyFolderTree(tmpRemovedTorrentData.savePathToRemove);
+
+    LogMsg(tr("'%1' was removed from the transfer list and hard disk.", "'xxx.avi' was removed...").arg(tmpRemovedTorrentData.name));
 }
 
 void Session::handleTorrentDeleteFailedAlert(libt::torrent_delete_failed_alert *p)
 {
+    const RemovedTorrentData tmpRemovedTorrentData = m_removedTorrents.take(p->info_hash);
     // libtorrent won't delete the directory if it contains files not listed in the torrent,
     // so we remove the directory ourselves
-    Utils::Fs::smartRemoveEmptyFolderTree(m_savePathsToRemove.take(p->info_hash));
+    Utils::Fs::smartRemoveEmptyFolderTree(tmpRemovedTorrentData.savePathToRemove);
+
+    LogMsg(tr("'%1' was removed from the transfer list but the files couldn't be deleted. Error: %2", "'xxx.avi' was removed...")
+           .arg(tmpRemovedTorrentData.name)
+           .arg(QString::fromLocal8Bit(p->error.message().c_str())), Log::CRITICAL);
```

Comment:
Correct way is `QString::fromStdString(p->error.message())`.

Reply:
Are you sure?
IIRC libtorrent populates these from the system using the system's locale. It isn't guaranteed that these strings are utf-8.

Reply:
Oops... I confused it with `libtorrent::alert::message()`.